### DISCUSSION TITLE
How to show all compile-time errors at once in the console, when building?

### DISCUSSION THREAD

**Discussion by Tobjoern** [vercel/next.js#48943] [Help]
### Summary

I really enjoy next.js, but one thing that has been bothering me for months,
is the way compile-time errors are handled.

Let's say I have a custom 'FormGroup' component in the 'shared' folder.
Now I moved this component from the 'shared' folder to the 'inputs' folder.
(I know auto-import-correction exists, but this example is just meant to illustrate some kind of component/api change, which affects multiple components)
When I run 'npm run dev', however, no compile-time errors appear. It is only once I navigate to a specific page,
where the error occurs, that I get a notification.
I find this behavior to be quite annoying, as one is only alerted by errors, once one tries to build/publish the app finally.

Furthermore when running 'npm run build', the compiler stops after the first occurrence of the error (in this case page1.tsx),
it doesn't even check page2.tsx.
Let's say I had used the 'FormGroup' component is 100 other components (which isn't uncommon at all), I would have
to run 'npm run build' 100 separate times, and fix each occurrence one by one.
See the example here: https://github.com/Tobjoern/nextjs-errors-demo

I'm still relatively new to next.js, so maybe I'm missing something, but I find the way nuxt.js handles this to be a lot better.
I created the same example here: https://github.com/Tobjoern/nuxtjs-error-demo
When I run 'npm run dev' it immediately shows me ALL the errors I have in the console, I can easily jump to the individual files,
in order to fix them.

Is there some setting I don't know about?
I'd be happy to hear your thoughts!

**Comment by mdalishanali** [comment]
Guys did you find any solutions?

**Reply by icyJoseph** [reply]
There's an undocumented feature... I think,

if you run the build with this env `__NEXT_TEST_MODE=true`, it'll log a lot more.

<img width="1716" alt="Screenshot 2024-04-17 at 08 48 42" src="https://github.com/vercel/next.js/assets/21013447/d8490612-5619-49e8-9ed2-cb785c70267f">

Put it behind something like:

```json
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint",
    "build:all-ts-errors": "__NEXT_TEST_MODE=true yarn build"
  },
```

Source: https://github.com/vercel/next.js/blob/canary/packages/next/src/lib/typescript/runTypeCheck.ts#L100

## EDIT

I looked around, and this `__NEXT_TEST_MODE` will make the build opt-in to a few code branches, that perhaps you don't want in production. I'd say, use it, **ONLY**, to get all the errors at once during development, but **_NOT TO DEPLOY_** your application. Alternatively just run `tsc` from the root of your project, though in monorepos that can sometimes be a bit annoying to get right.

**Reply by maheshguptha49** [reply]
Thank you so much 
I am migrating to next 13 from 12 and this command saved lot of time.
